kafka-ui-react-app/src/components/common/JSONEditor/JSONEditor.tsx

```diff
@@ -1,22 +1,29 @@
 /* eslint-disable react/jsx-props-no-spreading */
 import AceEditor, { IAceEditorProps } from 'react-ace';
 import 'ace-builds/src-noconflict/mode-json5';
+import 'ace-builds/src-noconflict/mode-protobuf';
 import 'ace-builds/src-noconflict/theme-tomorrow';
+import { SchemaType } from 'generated-sources';
 import React from 'react';
 import ReactAce from 'react-ace/lib/ace';
 import styled from 'styled-components';
 
 interface JSONEditorProps extends IAceEditorProps {
   isFixedHeight?: boolean;
+  schemaType?: string;
 }
 
 const JSONEditor = React.forwardRef<ReactAce | null, JSONEditorProps>(
   (props, ref) => {
-    const { isFixedHeight, ...rest } = props;
+    const { isFixedHeight, schemaType, ...rest } = props;
     return (
       <AceEditor
         ref={ref}
-        mode="json5"
+        mode={
+          schemaType === SchemaType.JSON || schemaType === SchemaType.AVRO
+            ? 'json5'
+            : 'protobuf'
+        }
         theme="tomorrow"
         tabSize={2}
         width="100%"
```

Comment:
I think @workshur mentioned that better to make separate components for `JSONEditor` and `ProtobufEditor` just to make easier to change editor from `AceEditor` to something else

I suggest to make separate `ProtobufEditor`

And you can add something like `SchemaEditor` which will decide which editor to use by `schemaType`

It seems your PR resolves #1415  

If you won't separate editors you have to, at least, rename it from `JSONEditor` to something else

Reply:
@Motherships if I will create new ProtobufEditor component the only difference from JSONEditor will be mode.
For proto type it is 'protobuf' for avro and json type it is 'json5'
I think there is no need to add a separate component just for that, I will rename JSONEditor to SchemaEditor, is it ok for you?

Reply:
@NelyDavtyan yes, and one more import for this mode

We have `SQLEditor` too, which only has different mode and theme (and has to import their mode and theme files)

I have to mention that I asked exactly this question (that the only thing that changes is mode and theme and if we make one component for each case it will bloat our build with unused imports) @workshur and I suggest to ask his opinion on this


I don`t know about `SchemaEditor` name in context where it will be used in all instances where we need to edit `json`

Just `Editor` will be fine in your case imo